src/Parser/Core/Modules/CastEfficiency.js

```diff
@@ -101,16 +151,20 @@ class CastEfficiency extends Analyzer {
     const cooldown = ability.castEfficiency.disabled ? null : ability.cooldown;
     const cooldownMs = !cooldown ? null : cooldown * 1000;
     const cdInfo = this._getCooldownInfo(ability);
+    const timeSpentCasting = cooldown && ability.charges < 2 ? this._getTimeSpentCasting(ability) : 0;
```

Comment:
When a ternary has multiple conditions it's often unclear if this results in either:
``` 
(cooldown && ability.charges < 2) ? this._getTimeSpentCasting(ability) : 0
```
or 
```
cooldown && (ability.charges < 2 ? this._getTimeSpentCasting(ability) : 0)
```

It requires advanced language knowledge to figure out the behavior of such a statement (&& taking priority over ternaries, which is different in some other languages). The fix also has little cost (imo improves readability), therefore it's recommended to use parenthesis in ternaries when you're checking more than one thing.

Reply:
Why are spells with multiple charges excluded? I reckon if Healing Rain had 2 charges you'd still want to add the casting time when it first goes on cooldown.

Reply:
The reason spells with charges is not included is because you can always have one of them recharging if you play optimally, except for, as you mention, the very first cast. It has a pretty minor impact overall though, and precasting would mess with it, so I'm not sure if it should be included.